src/wasm-interpreter.h

```diff
@@ -75,11 +75,45 @@ typedef std::vector<Literal> LiteralList;
 
 // Debugging helpers
 #ifdef WASM_INTERPRETER_DEBUG
-#define NOTE_ENTER(x) { std::cout << "visit " << x << " : " << curr << "\n"; }
-#define NOTE_ENTER_(x) { std::cout << "visit " << x << "\n"; }
-#define NOTE_NAME(p0) { std::cout << "name " << '('  << Name(p0) << ")\n"; }
-#define NOTE_EVAL1(p0) { std::cout << "eval " << '('  << p0 << ")\n"; }
-#define NOTE_EVAL2(p0, p1) { std::cout << "eval " << '('  << p0 << ", " << p1 << ")\n"; }
+class Indenter {
+static int indentLevel;
+
+const char* _str;
+
+public:
+  Indenter(const char* str) : _str(str) {
+    ++indentLevel;
+  }
+  ~Indenter() {
+    print();
+    std::cout << "exit " << _str << '\n';
+    --indentLevel;
+  }
+
+  static void print() {
+    std::cout << indentLevel << ':';
+    for (int i = 0; i <= indentLevel; ++i) {
+      std::cout << ' ';
+    }
+  }
+};
+int Indenter::indentLevel = 0;
```

Comment:
The indentation level needs to live across instances. The class takes advantage of RAII to decrement the indentation when we're done visiting a node. So, a given instance is transient, but the level of indentation is fundamentally shared.

For now I can add `wasm-interpreter.cpp` with the implementation of the `Indenter` class, and as a future project we can start migrating things into it.